SimG4Core/Application/src/StackingAction.cc

```diff
@@ -260,7 +260,8 @@ G4ClassificationOfNewTrack StackingAction::ClassifyNewTrack(const G4Track * aTra
 	  ((pdg == 2112) && (ke < kmaxNeutron))) { classification = fKill; }
     }
     if (!trackNeutrino  && classification != fKill) {
-      if (pdg == 12 || pdg == 14 || pdg == 16 || pdg == 18) 
+      if (pdg == 12 || pdg == 14 || pdg == 16 || pdg == 18 ||
```

Comment:
The old code used abs - but we would like to distinguish proton and anti-proton. So we need abs for something and not for some other. We can define apdg=abs(pdg) and use that. But I am not sure which is better
________________________________
From: David Lange [notifications@github.com]
Sent: 24 August 2016 18:17
To: cms-sw/cmssw
Cc: Sunanda Banerjee; Mention
Subject: Re: [cms-sw/cmssw] bsunanda:Run2-sim01 Checking neutrino tracking and saving info (#15572)


In SimG4Core/Application/src/StackingAction.cc<https://github.com/cms-sw/cmssw/pull/15572#discussion_r76089118>:

> @@ -260,7 +260,8 @@ G4ClassificationOfNewTrack StackingAction::ClassifyNewTrack(const G4Track * aTra
>         ((pdg == 2112) && (ke < kmaxNeutron))) { classification = fKill; }
>      }
>      if (!trackNeutrino  && classification != fKill) {
> -      if (pdg == 12 || pdg == 14 || pdg == 16 || pdg == 18)
> +      if (pdg == 12 || pdg == 14 || pdg == 16 || pdg == 18 ||


better to take abs(pdg) instead of doubling the if statements
@bsunanda<https://github.com/bsunanda>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/15572/files/72e2e248a14795f64470f4ddc0292e0a04f0183c#r76089118>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AEzMuhJXtaYqo5jAkAWPxSzndiErpotfks5qjG6KgaJpZM4JrSIG>.